File: Selenium/Activities/Activity10.py
from selenium import webdriver
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as fireFoxService
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
import time
from selenium.webdriver.support import expected_conditions as EC
service=fireFoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=service) as driver:
    driver.get("https://training-support.net/selenium/dynamic-controls")
    print("Page Title:- " + driver.title)

    checkBx=driver.find_element(By.ID,"dynamicCheckbox")
    wait = WebDriverWait(driver, 10)

    if checkBx.is_displayed():
        print("CheckBox Displayed")

    else:
        print("CheckBox Not Displayed")

    driver.find_element(By.ID,"toggleCheckbox").click()
    
    wait.until(EC.invisibility_of_element(checkBx))
    if checkBx.is_displayed():
        print("CheckBox Displayed")

    else:
        print("CheckBox Not Displayed")

    driver.close()

    
    

File: Selenium/Activities/Activity10.java
package example;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.interactions.Actions;



public class Activity10 {

	public static void main(String[] args) {

		WebDriver driver = new ChromeDriver();
		driver.get("https://training-support.net/webelements/mouse-events");
		System.out.println("Title of the page is: " + driver.getTitle());
		
		WebElement cargoLock = driver.findElement(By.xpath("//h1[text()='Cargo.lock']"));
        WebElement cargoToml = driver.findElement(By.xpath("//h1[text()='Cargo.toml']"));
        WebElement src = driver.findElement(By.xpath("//h1[text()='src']"));
        WebElement target = driver.findElement(By.xpath("//h1[text()='target']"));
		
        Actions action = new Actions(driver);
		action.click(cargoLock).pause(1000).moveToElement(cargoToml).pause(1000).click().build().perform();
		String text = driver.findElement(By.id("result")).getText();
		System.out.println("Text after clicking Cargo.toml: " + text);
		action.doubleClick(src).build().perform();
		action.contextClick(target).build().perform();
		
		action.click(driver.findElement(By.xpath("//span[text()='Open']"))).build().perform();
		String Message = driver.findElement(By.id("result")).getText();
        System.out.println(Message);
        driver.quit();
		
		
	}

}
